go/plugins/googlegenai/gemini.go

```diff
@@ -458,42 +326,44 @@ func generate(
 	}
 	r.Request = input
 	if cache != nil {
-		r.Message.Metadata = setCacheMetadata(r.Message.Metadata, cache)
+		r.Message.Metadata = cacheMetadata(r.Message.Metadata, cache)
 	}
 
 	return r, nil
 }
 
-// toGeminiRequest translates from [*ai.ModelRequest] to
-// *genai.GenerateContentParameters
+// toGeminiRequest translates an [*ai.ModelRequest] to
+// *genai.GenerateContentConfig
 func toGeminiRequest(input *ai.ModelRequest, cache *genai.CachedContent) (*genai.GenerateContentConfig, error) {
-	gcc := genai.GenerateContentConfig{
-		CandidateCount: 1,
-	}
-
-	c, err := configFromRequest(input)
+	gcc, err := configFromRequest(input)
 	if err != nil {
 		return nil, err
 	}
 
-	// Convert standard fields
-	if c.MaxOutputTokens != 0 {
-		gcc.MaxOutputTokens = int32(c.MaxOutputTokens)
+	// only one candidate is supported by default
+	gcc.CandidateCount = 1
+
+	// Genkit primitive fields must be used instead of go-genai fields
+	// i.e.: system prompt, tools, cached content, response schema, etc
+	if gcc.SystemInstruction != nil {
```

Comment:
@apascal07, here's where we validate the user request message do not set Genkit native features. I've added:
- System prompts
- Tools (just for function declaration tools, but kept the ability to support Code Execution, Google Search and so on)
- Cached  contents
- Response schema
- Response mime type